spec/services/service_result_spec.rb

```diff
@@ -151,6 +151,38 @@
         expect(instance.errors).not_to eq result.errors
       end
     end
+
+    context "when providing active record result" do
+      let(:result) { build(:work_package) }
+
+      it "creates a new errors instance" do
+        instance = described_class.new(result:)
+        base = instance.errors.instance_eval { @base } # rubocop:disable RSpec/InstanceVariable
+        expect(base).to eq(result)
+      end
+    end
+
+    context "when providing some other result" do
+      let(:result) { "wat?" }
+
+      it "creates a new errors instance" do
+        instance = described_class.new(result:)
+        base = instance.errors.instance_eval { @base } # rubocop:disable RSpec/InstanceVariable
+        expect(base).to eq(instance)
+      end
+    end
+
+    context "when using default errors from result" do
+      let(:result) { build(:work_package) }
+
+      it "uses the result's human_attribute_name" do
+        instance = described_class.new(result:)
+        instance.errors.add(:base, "some error")
+        instance.errors.add(:foo, "an error for foo")
+
+        expect { instance.errors.full_messages }.not_to raise_error
+      end
+    end
```

Comment:
When adding another test like in the other PR:

```suggestion
    context "when result is an ActiveRecord object" do
      let(:result) { build(:work_package) }

      it "errors.full_messages works and uses the result's human_attribute_name" do
        instance = described_class.new(result:)
        instance.errors.add(:base, "some error")
        instance.errors.add(:foo, "an error for foo")

        expect { instance.errors.full_messages }.not_to raise_error
      end
    end

    context "when result is not an ActiveRecordobject" do
      let(:result) { "I am a result" }

      it "errors.full_messages works" do
        instance = described_class.new(result:)
        instance.errors.add(:base, "some error")
        instance.errors.add(:foo, "an error for foo")

        expect { instance.errors.full_messages }.not_to raise_error
      end
    end
```

Then this test does not pass. I still does a "NoMethodError: undefined method 'human_attribute_name' for class ServiceResult"

(and I changed the test description because I was not understanding the previous ones)

Reply:
Yes, this is expected. This PR does not fix this error. Rather, it ensures to always use the result as a base if it responds_to errors. In case you pass a result that does not implement AM::Errors, the other PR is still needed.